src/windows/minidump_writer.rs

```diff
@@ -0,0 +1,279 @@
+    /// In the case of a `STATUS_INVALID_HANDLE` exception, this function
+    /// enumerates all of the handle operations that occurred within the crashing
+    /// process and fills out a minidump user stream with the ops pertaining to
+    /// the last invalid handle that is enumerated, which is, presumably
+    /// (hopfully?), the one that led to the exception
+    fn fill_handle_stream(&self) -> Option<(Vec<u8>, md::MINIDUMP_USER_STREAM)> {
+        if self.crash_context.exception_code != STATUS_INVALID_HANDLE {
+            return None;
+        }
+
+        // State object we pass to the enumeration
+        struct HandleState {
+            ops: Vec<av::AVRF_HANDLE_OPERATION>,
+            last_invalid: u64,
+        }
+
+        unsafe extern "system" fn enum_callback(
+            resource_description: *mut c_void,
+            enumeration_context: *mut c_void,
+            enumeration_level: *mut u32,
+        ) -> u32 {
+            let description = &*resource_description.cast::<av::AVRF_HANDLE_OPERATION>();
+            let mut hs = &mut *enumeration_context.cast::<HandleState>();
+
+            // Remember the last invalid handle operation.
+            if description.OperationType == av::OperationDbBADREF as u32 {
+                hs.last_invalid = description.Handle;
+            }
+
+            // Record all handle operations.
+            hs.ops.push(*description);
+            *enumeration_level = av::HeapEnumerationEverything as u32;
+            ERROR_SUCCESS
+        }
+
+        let mut hs = HandleState {
+            ops: Vec::new(),
+            last_invalid: 0,
+        };
+
+        // https://docs.microsoft.com/en-us/windows/win32/api/avrfsdk/nf-avrfsdk-verifierenumerateresource
+        // SAFETY: syscall
+        if unsafe {
+            av::VerifierEnumerateResource(
+                self.crashing_process,                // process to enumerate the handles for
+                0,                                    // flags
+                av::AvrfResourceHandleTrace,          // resource typea, we want to trace handles
+                Some(enum_callback),                  // enumeration callback
+                (&mut hs as *mut HandleState).cast(), // enumeration context
+            )
+        } != 0
+        {
+            let mut stream_buf = Vec::new();
+
+            // https://docs.microsoft.com/en-us/windows/win32/api/minidumpapiset/ns-minidumpapiset-minidump_handle_operation_list
+            let mut md_list = md::MINIDUMP_HANDLE_OPERATION_LIST {
+                SizeOfHeader: std::mem::size_of::<md::MINIDUMP_HANDLE_OPERATION_LIST>() as u32,
+                SizeOfEntry: std::mem::size_of::<av::AVRF_HANDLE_OPERATION>() as u32,
+                NumberOfEntries: 0,
+                Reserved: 0,
+            };
+
+            stream_buf.resize(md_list.SizeOfHeader as usize, 0);
+
+            #[inline]
+            fn to_bytes<T: Sized>(v: &T) -> &[u8] {
+                let s = std::slice::from_ref(v);
```

Comment:
I don't think the first part is necessary - A single expression of `std::slice::from_raw_parts((v as *const T) as *const u8, std::mem::size_of::<T>())` seems a bit more idiomatic?